doc/neps/nep-0028-deprecation_policy.rst

```diff
@@ -0,0 +1,184 @@
+==========================================================================================
+NEP 28 - A standard community policy for dropping support of old Python and NumPy versions
+==========================================================================================
+
+
+:Author: Thomas A Caswell <tcaswell@gmail.com>, Andreas Mueller, Brian Granger, Madicken Munk, Ralf Gommers, Matt Haberland <mhaberla@calpoly.edu>, Matthias Bussonnier <bussonniermatthias@gmail.com>, Stefan van der Walt
+:Status: Draft
+:Type: Informational Track
+:Created: 2019-07-13
+
+
+Abstract
+--------
+
+All projects across the ecosystem should adopt a common time window
+based policy for increasing the minimum version of Python and NumPy
+that downstream projects support.  By standardizing this policy
+across community we will make it easier for down stream projects to
+plan.
+
+This is an unusual NEP in that it is recommendations for community
+wide policy and not for changes to NumPy itself.  However, we do not
+currently have a place for SPEEPs (Scientific Python Ecosystem
+Enhancement Proposals) and given NumPy's central roll in our community
+this seems like the proper place to document this.
+
+
+This is being put forward by maintainers of Matplotlib, scikit-learn,
+IPython, Jupyter, yt, SciPy, NumPy, and scikit-image.
+
+
+
+Detailed description
+--------------------
+
+When making a new major or minor release, projects should support all
+minor versions of ``CPython`` initially released in the prior 42
+months from their anticipated release date and all minor version of
+NumPy released in the prior 24 months.
+
+
+The diagram::
+
+       Jan 16      Jan 17      Jan 18      Jan 19      Jan 20
+       |           |           |           |           |
+  +++++|+++++++++++|+++++++++++|+++++++++++|+++++++++++|++++++++++++
+   |              |                  |               |
+   py 3.5.0       py 3.6.0           py 3.7.0        py 3.8.0
+  |-----------------------------------------> Feb19
+            |-----------------------------------------> Dec19
+                      |-----------------------------------------> Nov20
+
+shows the support windows for ``CPython``.  A project with a minor
+release in Feb19 should support py35 and newer, a project with a minor
+release in Dec19 should support py36 and newer, and a project
+with a minor release in Nov20 should support py37 and newer.
+
+The current CPython release cadence is 18 months so a 42 month window
```

Comment:
Just an FYI. It is highly likely that the release cadence will become shorter than 18 months for CPython.

Reply:
All the more reason for this!